td-shim/src/bin/td-shim/tcg.rs

```diff
@@ -34,7 +42,20 @@ impl TdEventLog {
             last: 0,
             started: false,
             truncated: false,
-        }
+        };
+
+        // Create the TCG_EfiSpecIDEvent as the first event
+        let first = TcgEfiSpecIdevent::default();
+        td_event_log
+            .log_event(
+                0,
+                EV_NO_ACTION,
+                first.as_bytes(),
+                &[0u8; SHA384_DIGEST_SIZE],
```

Comment:
The digest for SpecidEvent is always 20 bytes - to align with SHA1 for compatibility reason.